src/frontend-samples/explode-sample/ExplodeApp.tsx

```diff
@@ -0,0 +1,337 @@
+      const transform = placement.transform;
+      const box = Range3d.create(element.bBoxHigh, element.bBoxLow);
+
+      element.transformWorld = transform;
+      element.boundingBox = transform.multiplyRange(box);
+
+      data.push(element);
+      row = await results.next();
+    }
+
+    return data;
+  }
+
+  /** Reads the response for tile content and returns the Tile Content. */
+  private static async readResponse(vp: Viewport, response: TileRequest.ResponseData): Promise<TileContent> {
+    const stream = new ByteStream((response as Uint8Array).buffer);
+    const reader = ImdlReader.create(stream, vp.iModel, vp.iModel.iModelId!, vp.view.is3d(), IModelApp.renderSystem);
+
+    let content: TileContent = { isLeaf: true };
+    if (reader) {
+      try {
+        content = await reader.read();
+      } catch (_) {
+        //
+      }
+    }
+    return content;
+  }
+}
+
+/** This provider both hides the original graphics of the element and inserts the transformed graphics. */
+class ExplodeProvider implements TiledGraphicsProvider, FeatureOverrideProvider {
+  private static _instances = new Map<number, ExplodeProvider>();
+  public data: ElementData[] = [];
+  /** Returns a provider associated with a given viewport. If one does not exist, it will be created. */
+  public static getOrCreate(vp: Viewport) {
+    let provider = ExplodeProvider._instances.get(vp.viewportId);
+    if (!provider)
+      provider = ExplodeProvider.createAndAdd(vp);
+    return provider;
+  }
+  /** Creates a provider associated with a given viewport. */
+  public static createAndAdd(vp: Viewport) {
+    const provider = new ExplodeProvider(vp);
+    ExplodeProvider._instances.set(vp.viewportId, provider);
+    provider.add(vp);
+    return provider;
+  }
+  /** Adds provider from viewport */
+  public add(vp: Viewport) {
+    if (!vp.hasTiledGraphicsProvider(this)) {
+      vp.addTiledGraphicsProvider(this);
+      vp.addFeatureOverrideProvider(this);
+    }
+  }
+  /** Drops provider from viewport */
+  public drop() {
+    this.vp.dropFeatureOverrideProvider(this);
+    this.vp.dropTiledGraphicsProvider(this);
+  }
+  /** Singles the viewport to redraw graphics. */
+  public invalidate() {
+    this.vp.setFeatureOverrideProviderChanged();
+  }
+  private constructor(public vp: Viewport) { }
+
+  public addFeatureOverrides(overrides: FeatureSymbology.Overrides, _vp: Viewport): void {
+    const app = FeatureAppearance.fromTransparency(1);
+    this.data.forEach((element) => {
+      // TODO: hide elements when Emphasized (not isolated)
+      overrides.overrideElement(element.id, app, true);
+    });
+  }
+
+  /** Apply the supplied function to each [[TileTreeReference]] to be drawn in the specified [[Viewport]]. */
+  public forEachTileTreeRef(viewport: ScreenViewport, func: (ref: TileTreeReference) => void): void {
+    // TODO: update tile tree with new tiles
+    viewport.view.forEachTileTreeRef(func);
```

Comment:
You are supposed to invoke `func` on the tile tree references owned by **this** provider, not those belonging to the view.